tgui/packages/tgui/interfaces/common/sorting/SortableTable/utils.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license ISC
+ */
+
+import { SortDirection } from "../type";
+import { SearchState, SortableTableHeaderConfig, SortableTableRowProps, SortState } from "./type";
+
+export const onSortClick = (index: number, current: SortState, setCallback: (nextState: SortState) => void) => {
+  if (current !== null) {
+    if (current.index === index) {
+      setCallback({
+        dir: (current.dir === SortDirection.Asc ? SortDirection.Desc : SortDirection.Asc),
+        index: index,
+      });
+    } else {
+      setCallback({ dir: SortDirection.Asc, index: index });
+    }
+  } else {
+    setCallback({ dir: SortDirection.Asc, index: index });
+  }
```

Comment:
```suggestion
  if (current && current.index === index ) {
    setCallback({
      dir: (current.dir === SortDirection.Asc ? SortDirection.Desc : SortDirection.Asc),
      index,
    });
  } else {
    setCallback({ dir: SortDirection.Asc, index });
  }
```
N.B. `index: index` -> `index` shorthand.